### PR TITLE
Class annotation to trace lambda methods

### PR DESCRIPTION
**Pull request by GDownes**
### Overview
New class annotation TraceLambda. Marking a class with the TraceLambda annotation will instruct the New Relic agent to automatically annotate lambda methods within the marked class with a Trace annotation. 

### Testing
Added unit tests to cover TraceLambda usage scenarios.
Deploy updated agent locally and reviewed traces in New Relic One.

### Checks

[x] Are your contributions backwards compatible with relevant frameworks and APIs? Yes
[x] Does your code contain any breaking changes? Please describe. No
[x] Does your code introduce any new dependencies? Please describe. No

### CODE REVIEW

File: newrelic-agent/src/test/java/com/newrelic/agent/instrumentation/classmatchers/TraceLambdaVisitorTest.java [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+
+    private void validateTraceAnnotations(InstrumentationContext context, String methodName, String metricName) {
+        Map<Method, TraceDetails> traceAnnotations = context.getTraceInformation().getTraceAnnotations();
+        Assert.assertFalse(traceAnnotations.isEmpty());
+        Map.Entry<Method, TraceDetails> traceAnnotation = traceAnnotations.entrySet().iterator().next();
+        Assert.assertEquals(methodName, traceAnnotation.getKey().getName());
+        Assert.assertEquals(metricName, traceAnnotation.getValue().metricName());
+    }
+}
+
```
Comment (XiXiaPdx) [inline newrelic-agent/src/test/java/com/newrelic/agent/instrumentation/classmatchers/TraceLambdaVisitorTest.java:58]: In my local testing, I tried a poorly named method called `notlambda$doingWork$1()`.   It's plain method, no lambdas.  

As expected, this did not match 😄   Would there be value in a test that we don't match on just `lambda`?

**Review by XiXiaPdx — APPROVED** [review]
@GDownes   LGTM! I had one curiosity regarding the tests and I provided a comment for it.

**Comment by XiXiaPdx** [comment]
Running AITs internally, hold please

File: newrelic-agent/src/test/java/com/newrelic/agent/instrumentation/classmatchers/TraceLambdaVisitorTest.java (hunk 1 above)
Comment (GDownes) [reply newrelic-agent/src/test/java/com/newrelic/agent/instrumentation/classmatchers/TraceLambdaVisitorTest.java:58]: Hi Xi,
The default behaviour is to run the following regex "^\\$?(lambda|anonfun)\\$(?<name>.*)"  against the names of all the static methods in the marked class. Only static methods with names matching the pattern will match. The default pattern aims to cover Java and Scala lambda expressions as they are compiled to static methods.

**Comment by XiXiaPdx** [comment]
Internal AITs look good, merging - https://javaagent-build.pdx.vm.datanerd.us/job/JavaAgent_PR_AIT/68/

**Pull request merged**